app/services/users/omniauth.rb

```diff
@@ -14,18 +14,31 @@ def get
   delegate :provider, :uid, :info, to: :auth
   delegate :email, to: :info
 
+  def authorization
+    authorization = Authorization.where(provider: provider, uid: uid).first
+    authorization ||= Authorization.new(provider: provider, uid: uid)
+    authorization.user
+  end
+
   def user
-    user = User.where(provider: provider, uid: uid, email: email).first
-    user ||= User.new(
-      provider: provider,
-      uid: uid,
-      email: email,
-      password: Devise.friendly_token[0, 20]
-    )
+    user = User.where(email: email).first
     user_create(user)
   end
 
+  def create_user_for_authorization
+    if authorization.blank?
+      authoriz = authorization
+      authoriz.user = user
+      authoriz.save
+    end
+    authorization
+  end
+
   def user_create(user)
+    user ||= User.new(
+      email: email,
+      password: Devise.friendly_token[0, 20]
+    )
```

Comment:
@marderer good attempt 👍 
I have suggestions left for another PR that can make your knowledge wider, [check](https://github.com/sinpai/datarockets-finance-app/pull/11#discussion_r174719678)

Reply:
@marderer why didn't you updated tests for this service?

Reply:
@AleksSenkou In the next [PR.](https://github.com/marderer/personal-finance/pull/33)